### ISSUE TITLE
Create unique names for duplicate layer names

### ISSUE THREAD

**Issue by mattmattvanvoorst** [react-spring/gltfjsx#28]
Hi, great tool first of all!

I'm trying to convert an entire scene but I'm noticing that models are partially breaking or even being displaced. When loading the .glb model through a regular gltf loader the model stays intact (apart from some bitmap textures but that is a different issue).

I'm seeing tons of duplicate types and layer names, and I have a suspicion because these materials and meshes do not have unique names this is causing me issues.

Would it be an idea to generate unique names for meshes and materials?

This is the .glb loaded through a .gltf loader
![image](https://user-images.githubusercontent.com/35974629/90250898-23031f80-de3d-11ea-8185-baec7886893b.png)

When generated though gltfjsx. No compression, precision on a 100, with types (apologies for the different perspectives)
![image](https://user-images.githubusercontent.com/35974629/90250935-331aff00-de3d-11ea-8be1-f6a54cc947a1.png)

These are some of the types that are generated.
![Screenshot 2020-08-14 at 14 49 24](https://user-images.githubusercontent.com/35974629/90251016-59d93580-de3d-11ea-8401-93605f5d161e.png)

I fully understand that the naming issue should be solved from the model, but I'm just wondering if we could tacked this through the tool as well.

**Comment by drcmda** [comment]
yes that's the one thing that kills gltfjsx, duplicates. blender thankfully disallows it and threejs creates unique names, but, unfortunately, the gltf spec allows it and if you have or made a gltf that's formed like that gltfjsx has no chance to do build a proper graph for you automatically.

if you can, you could append the gltf here and i can see what's up - but the parsing is done by threejs own gltfloader so chances are i wont be able to fix it unless it's something else that's not right.

you can always form your graph by hand of course, if you know where meshes and materials are

```jsx
const { scene } = useLoader()
return (
  <mesh geometry={scene.children[0].geometry} ... />
    <mesh geometry={scene.children[0].children[0].geometry} ... />
      <mesh geometry={scene.children[0].children[0].children[0].geometry} ... />
```

but it feels kind of shaky. the optimal solution would be to give the gltf proper names, you can use your plain editor.

**Comment by mattmattvanvoorst** [comment]
We ended up doing just that; we imported/exported the gltf through Blender, and Blender updated all objectnames to unique ones. That fixed all our issues.

Over at GLTFloader there are some proposals for de-duplicating names similarly like Blender does. It might be worth keeping an eye on that, and as soon as they roll out their new version update the one in gltfjsx.

Issue being brought up:
https://github.com/mrdoob/three.js/issues/15087

Proposal mentioned in issue:
https://github.com/mrdoob/three.js/pull/16639

I'll close this issue as this is not a problem with gltfjsx, but an issue with the model itself that GLTFLoader could help in fixing.
